### PR TITLE
osd/OSDMap: Clean up: do not call get_max_osd everytime during for-loops

### PR DESCRIPTION
**Pull request by shinobu-x**
Prepare storage for ``get_max_osd()`` before going into for-loops.
Replace post-increment with pre-increment in for statement.

mon: OSDMonitor.cc
osd: OSDMap.cc

Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by shinobu-x** [comment]
retest this please

**Comment by shinobu-x** [comment]
retest this please

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -3491,9 +3504,12 @@ bool OSDMonitor::preprocess_command(MonOpRequestRef op)
       if (!f)
 	ds << " ";
     } else if (prefix == "osd ls") {
+
+//      int max_osds = osdmap.get_max_osd();
+
```
Comment (liewegas) [inline src/mon/OSDMonitor.cc]: remove this?

File: src/mon/OSDMonitor.cc [hunk 2]
```diff
@@ -6718,7 +6735,7 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
       // Check if the OSDs exist between current max and new value.
       // If there are any OSDs exist, then don't allow shrinking number
       // of OSDs.
-      for (int i = newmax; i < osdmap.get_max_osd(); i++) {
+      for (int i = newmax; i < osdmap.get_max_osd(); ++i) {
```
Comment (liewegas) [inline src/mon/OSDMonitor.cc]: not here?

File: src/osd/OSDMap.cc [hunk 3]
```diff
@@ -2329,6 +2329,9 @@ void OSDMap::dump_erasure_code_profiles(const map<string,map<string,string> > &p
 
 void OSDMap::dump(Formatter *f) const
 {
+
+  int max_osds = get_max_osd();
+
```
Comment (liewegas) [inline src/osd/OSDMap.cc]: no need for blank lines

File: src/osd/OSDMap.cc [hunk 4]
```diff
@@ -2338,7 +2341,7 @@ void OSDMap::dump(Formatter *f) const
   f->dump_float("nearfull_ratio", nearfull_ratio);
   f->dump_string("cluster_snapshot", get_cluster_snapshot());
   f->dump_int("pool_max", get_pool_max());
-  f->dump_int("max_osd", get_max_osd());
+  f->dump_int("max_osd", max_osds);
```
Comment (liewegas) [inline src/osd/OSDMap.cc]: and we only call it once, so the original code is fine

**Comment by shinobu-x** [comment]
@liewegas Thanks for taking your precious time :) During testing calling function returning integer (let's say N) in loop statement, this function returned N times. So I made this PR.

It's really up to you if this PR is worth being merged or not. Thanks again!

**Comment by shinobu-x** [comment]
retest this please

**Comment by shinobu-x** [comment]
retest this please

**Comment by shinobu-x** [comment]
retest this please

**Comment by shinobu-x** [comment]
@liewegas O.k, now I'm getting more clear about:
> The get_max_osds() is an accessor that returns a member variable, so this will compile down to the same code both before and after the change.

**Comment by shinobu-x** [comment]
retest this

**Comment by shinobu-x** [comment]
retest this please

**Comment by shinobu-x** [comment]
retest this please

**Comment by shinobu-x** [comment]
retest this please

**Comment by shinobu-x** [comment]
retest this please

**Comment by shinobu-x** [comment]
retest this please

**Comment by shinobu-x** [comment]
retest this please

**Pull request closed without merging**